draft-cardwell-ccwg-bbr.xml

```diff
@@ -1221,13 +1218,9 @@ sustain its achievable throughput.</t>
   /* Track ACK state and update BBR.max_bw window and
    * BBR.inflight_hi. */
   BBRAdaptUpperBounds():
-    if (BBR.ack_phase == ACKS_PROBE_STARTING and BBR.round_start)
-      /* starting to get bw probing samples */
-      BBR.ack_phase = ACKS_PROBE_FEEDBACK
-    if (BBR.ack_phase == ACKS_PROBE_STOPPING and BBR.round_start)
+    if (BBR.state == ProbeBW_DOWN and BBR.round_start and !rs.is_app_limited)
```

Comment:
I agree it would be great to simplify this ack_phase stuff... But AFAICT this has a bug: in a particular ProbeBW_DOWN episode, before a round trip in ProbeBW_DOWN elapses (before BBR.round_start becomes true),  BBRIsTimeToCruise() may return true if inflight is less than the estimated BDP. So this expression may never be true, and the BBRAdvanceMaxBwFilter() may never be called. WDYT?

Reply:
Good point. One way to avoid the bug is to advance the cycle when entering ProbeBW_DOWN if the transport isn't app limited. That's the approach I've taken in mvfst.